Project/Gem/Source/Vision/IdealVisionSystem.cpp

```diff
@@ -0,0 +1,305 @@
+            {
+                if (!m_excludeEntities.contains(result.m_entityId))
+                {
+                    const AZ::Aabb aabb = result.m_shape->GetAabbLocal();
+
+                    AZ::Transform resutlTransform;
+                    AZ::TransformBus::EventResult(resutlTransform, result.m_entityId, &AZ::TransformBus::Events::GetWorldTM);
+                    const AZ::Transform resultTransformLocal = sensorTransformInv * resutlTransform;
+
+                    AZStd::string targetName;
+                    AZ::ComponentApplicationBus::BroadcastResult(
+                        targetName, &AZ::ComponentApplicationBus::Events::GetEntityName, result.m_entityId);
+
+                    geometry_msgs::msg::Pose pose;
+                    pose.orientation = ROS2::ROS2Conversions::ToROS2Quaternion(resultTransformLocal.GetRotation());
+                    pose.position = ROS2::ROS2Conversions::ToROS2Point(resultTransformLocal.GetTranslation());
+                    poseArray.poses.push_back(pose);
+
+                    // construct 3D detection
+                    vision_msgs::msg::Detection3D detection3D;
+                    detection3D.header = header;
+                    detection3D.id = targetName.c_str();
+
+                    detection3D.bbox.center.position = pose.position;
+                    detection3D.bbox.size.x = aabb.GetXExtent() * 2.0f;
+                    detection3D.bbox.size.y = aabb.GetYExtent() * 2.0f;
+                    detection3D.bbox.size.z = aabb.GetZExtent() * 2.0f;
+
+                    detection3D.results.resize(1);
+                    detection3D.results.front().pose.pose = pose;
+                    detection3D.results.front().hypothesis.class_id = -1;
+                    detection3D.results.front().hypothesis.score = 1.0f;
+
+                    detection3DArray.detections.push_back(detection3D);
+
+                    // construct 2D detection
+                    vision_msgs::msg::Detection2D detection2D;
+                    detection2D.header = header;
+                    detection2D.id = targetName.c_str();
+
+                    AZ::Vector2 center2D = GetPointIn2D(resultTransformLocal.GetTranslation(), m_cameraMatrix);
+
+                    detection2D.results.resize(1);
+                    detection2D.results.front().pose.pose.position.x = center2D.GetX();
+                    detection2D.results.front().pose.pose.position.y = center2D.GetY();
+                    detection2D.results.front().hypothesis.class_id = -1;
+                    detection2D.results.front().hypothesis.score = 1.0f;
+
+                    detection2DArray.detections.push_back(detection2D);
+                }
+            }
+        }
+
+        m_detectionArrayPublisher->publish(poseArray);
+        m_detection2DPublisher->publish(detection2DArray);
+        m_detection3DPublisher->publish(detection3DArray);
+    }
+    void IdealVisionSystem::Visualise()
+    {
```

Comment:
Sensor Configuration has `visualise` flag, that can be disabled in editor, but IdealVision will be visualized regardless of this flag.


the following code could be used:

```
        if (!m_sensorConfiguration.m_visualise)
        {
            return;
        }
   ```